### DISCUSSION TITLE
<Show fallback={}> doesn't feel chronological

### DISCUSSION THREAD

**Discussion by birkskyum** [solidjs/solid#1903] [Ideas]
Using `<Show>` is a lot like using an if-statement.

```ts
<Show when={}>
// If when-condition is true
<Show>
```

When having a fallback, I'd hope to get a feeling of using an "if {} else {}" statement, like a ternary  ? : operator often used in JSX, but because the fallback has to be defined first, it ends up as an else {} if {}, which doesn't feel chronological, and porting a react app with many ternary operators to `<Show>` is quite daunting because it requires flipping the order of every single one, or negating all the conditions which likely will introduce bugs.

```ts
<Show when={} fallback={
// else
}>
// If when-condition is true
</Show>
```

A possible solution would be to add a content or template to `<Show>`

```ts
<Show when={} content={
// If condition is true
} fallback={
// else
} />
```

Or even make it the primitive ternary `<If>` operator it basically is.

```ts
<If cond={} then={
// If condition is true
} else={
// else
} />
```

**Comment by ryansolid** [comment]
I mean I could be that person and say you can do that already with children.

```jsx
<Show
  when={}
  children={}
  fallback={}
/>
```
It is interesting the idea of not using children purposefully and forcing a self-closing element.

**Reply by birkskyum** [reply]
Ah, clever! I'll use that.
